subprojects/docs/src/docs/userguide/upgrading_version_5.adoc

```diff
@@ -0,0 +1,105 @@
+// Copyright 2018 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[upgrading_version_5]]
+= Upgrading your build from Gradle 5.0
+
+This chapter provides the information you need to migrate your Gradle 5.0 builds to Gradle 5.1.
+
+We recommend the following steps for all users:
+
+. Try running `gradle help --scan` and view the https://gradle.com/enterprise/releases/2018.4/#identify-usages-of-deprecated-gradle-functionality[deprecations view] of the generated build scan.
++
+image::deprecations.png[Deprecations View of a Gradle Build Scan]
++
+This is so that you can see any deprecation warnings that apply to your build.
++
+Alternatively, you could run `gradle help --warning-mode=all` to see the deprecations in the console, though it may not report as much detailed information.
+. Update your plugins.
++
+Some plugins will break with this new version of Gradle, for example because they use internal APIs that have been removed or changed. The previous step will help you identify potential problems by issuing deprecation warnings when a plugin does try to use a deprecated part of the API.
++
+. Run `gradle wrapper --gradle-version 5.1` to update the project to 5.1
+. Try to run the project and debug any errors using the <<troubleshooting.adoc#troubleshooting, Troubleshooting Guide>>.
+
+////
+In addition, Gradle has added several significant new and improved features that you should consider using in your builds:
+
+*
+
+Other notable changes to be aware of that may break your build include:
+
+*
+////
+
+[[changes_5.1]]
+== Upgrading from 5.0 and earlier
+
+=== Potential breaking changes
+
+The changes in this section have the potential to break your build, but the vast majority have been deprecated for quite some time and few builds will be affected by a large number of them.
+
+The following changes were not previously deprecated:
+
+* Passing arguments to Windows Resource Compiler
+
++
+To follow idiomatic <<#lazy_configuration.adoc, Provider API>> practices, the `WindowsResourceCompile` task has been converted to use the Provider API.
++
+Passing additional compiler arguments now follow the same pattern as `CppCompile` and other tasks.
+
+////
+The following breaking changes will appear as deprecation warnings with Gradle 5.1:
```

Comment:
Not sure if this lengthy template is very helpful, though I agree with the concept of a "The following breaking changes will appear as deprecation warnings with Gradle 5.x" subsection.